mindsdb/integrations/handlers/huggingface_api_handler/huggingface_api_handler.py

```diff
@@ -16,118 +17,166 @@ class HuggingFaceInferenceAPIHandler(BaseMLEngine):
 
     name = 'huggingface_api'
 
+    @staticmethod
+    def create_validation(target, args=None, **kwargs):
+
+        if 'using' in args:
+            args = args['using']
+
+        hf_api = HfApi()
+
+        if 'model_name' not in args:
+            # detect model by task
+            task = args.get('task')
+            if task is None:
+                raise Exception('model_name or task have to be specified')
+
+            # get from task
+            config = ConfigParser()
+            model_name = config.get_config_dict()['TASK_MODEL_MAP'].get(task)
+            if model_name is None:
+                raise Exception(f'Model not found by task: {task}')
+
+            args['model_name'] = model_name
+        else:
+            # detect task by model
+            metadata = hf_api.model_info(args['model_name'])
```

Comment:
This is a really great check! I should have done this already, but I just added this check to the package, so that we don't have to maintain the `huggingface_hub` dependency here. Is that cool?